docs/api-reference/next/server.md

```diff
@@ -86,6 +86,21 @@ All methods above return a `NextResponse` object that only takes effect if it's
 import { NextResponse } from 'next/server'
 ```
 
+### Setting the cookie before a redirect
+
+In order to set the `cookie` _before_ a redirect, you can create an instance of `NextResponse`, then access the `cookie` method on the instance, before returning the response.
+
+```ts
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
+
+export function middleware(req: NextRequest, res: NextResponse) {
+  const res = NextResponse.redirect('/') // creates an actual instance
```

Comment:
If I remember correctly setting a cooking before a redirect has a bug in Chrome where the cookie is not set until you refresh again. Might be that we have to recommend a different approach 🤔

Reply:
@Timer @ijjk I hope one of you remembers if this was the case or not.

Reply:
https://bugs.chromium.org/p/chromium/issues/detail?id=696204

https://github.com/vercel/preview-mode-demo/blob/master/src/pages/api/sameorigin/%5BsnapshotId%5D.tsx

Reply:
I kept the code, but added a note about the bug and the potential consequences.